code/control/DNDeploymentDispatcher.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+use Finite\StateMachine\StateMachine;
+
+class DNDeploymentDispatcher extends Dispatcher {
+
+	const ALLOW_DNDEPLOYMENT_APPROVE = 'ALLOW_DNDEPLOYMENT_APPROVE';
+	const ALLOW_DNDEPLOYMENT_QUEUE = 'ALLOW_DNDEPLOYMENT_QUEUE';
+
+	private static $allowed_actions = [
+		'apply',
+		'canApply',
+	];
+
+	private static $url_handlers = [
+		'$Id/apply/$State' => 'apply',
+		'$Id/can/$State' => 'canApply',
+	];
+
+	public function getModel($name) {
+		$id = $this->request->param('Id');
+		$deployment = DNDeployment::get()->byId($id);
+		return [
+			'Status' => $deployment->Status
+		];
+	}
+
+	public function apply(SS_HTTPRequest $request) {
+		$this->checkSecurityToken();
+
+		$project = $this->getCurrentProject();
+		if(!$project) {
+			return $this->project404Response();
+		}
+
+		$env = $this->getCurrentEnvironment($project);
+		if(!$env) {
+			return $this->environment404Response();
+		}
+
+		$machine = $this->getMachine();
+		$machine->apply($request->param('State'));
```

Comment:
It will blow up in your face with exception ;-) I decided not to spend too much time here, because I don't have any actually use cases for this dispatcher yet. The whole API might be thrown away after deliberation.